packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ModelReaderWriterContextDefinition.cs

```diff
@@ -193,14 +209,14 @@ private void CollectBuildableTypesFromFrameworkType(
                     var csharpPropertyType = new CSharpType(propertyType);
                     var typeToCheck = csharpPropertyType.IsCollection ? csharpPropertyType.ElementType : csharpPropertyType;
 
-                    CollectBuildableTypesRecursive(typeToCheck.WithNullable(false), buildableTypes, visitedTypes, providers);
+                    CollectBuildableTypesRecursive(typeToCheck.WithNullable(false), null, visitedTypes, visitedTypeProviders, buildableProviders, buildableTypes);
```

Comment:
We shouldn't have to call in to the base recursive method if we know we are only dealing with framework types. Framework types would always have property that are themselves framework types, right?